internal/controller/management/projects/projects.go

```diff
@@ -341,6 +364,194 @@ func (r *reconciler) ensureProjectAdminPermissions(
 	return nil
 }
 
+func (r *reconciler) ensureDefaultProjectRoles(
+	ctx context.Context,
+	project *kargoapi.Project,
+) error {
+	logger := logging.LoggerFromContext(ctx).WithFields(log.Fields{
+		"project":   project.Name,
+		"name":      project.Name,
+		"namespace": project.Name,
+	})
+
+	const adminRoleName = "admin"
+	const viewerRoleName = "viewer"
```

Comment:
Should we consider naming these `kargo-admin` and `kargo-viewer` in case this clashes with someone installing Kargo into a cluster that is purposed for workloads as well?

Reply:
But these are in project namespaces. Do we ever expect workloads in project namespaces?

Reply:
It's not that I expect workloads in kargo namespaces/projects. Oftentimes, there's just some mechanism (like [kyverno](https://kyverno.io/docs/writing-policies/generate/)) which might auto-create stuff in namespace.

Since `admin` and `viewer` are common names, I suggested the `kargo-` prefix to avoid potential for someone installing kargo into a cluster that already has `admin` and/or `viewer` as a role that might get created in all namespaces.

This is not a hill I'd die on.

Reply:
Ah... ok... that makes sense, so happy to change this.